2011-02-08  Steve Langasek  <steve.langasek@ubuntu.com>
	* elf/Makefile: use the correct path in the ldd script as well

2006-01-21  Aurelien Jarno  <aurel32@debian.org> 

	* Makeconfig: new configuration variable rtlddir to change the
	  dynamic linker in the ELF binaries, which defaults to slibdir
	  
2005-08-30  Andreas Jochen  <Andreas Jochens>
	* Makeconfig: use rtlddir instead of slibdir
	* elf/Makefile: likewise

---
 Makeconfig   |    7 ++++++-
 elf/Makefile |    4 ++--
 2 files changed, 8 insertions(+), 3 deletions(-)

--- a/Makeconfig
+++ b/Makeconfig
@@ -202,6 +202,11 @@
 endif
 inst_slibdir = $(install_root)$(slibdir)
 
+# Directory of the dynamic linker in the ELF binaries
+ifndef rtlddir
+rtlddir := $(slibdir)
+endif
+
 # Prefix to put on files installed in $(libdir).  For libraries `libNAME.a',
 # the prefix is spliced between `lib' and the name, so the linker switch
 # `-l$(libprefix)NAME' finds the library; for other files the prefix is
@@ -486,7 +491,7 @@
 endif
 ifndef config-LDFLAGS
 ifeq (yesyes,$(build-shared)$(elf))
-config-LDFLAGS = -Wl,-dynamic-linker=$(slibdir)/$(rtld-installed-name)
+config-LDFLAGS = -Wl,-dynamic-linker=$(rtlddir)/$(rtld-installed-name)
 endif
 endif
 ifndef link-libc
--- a/elf/Makefile
+++ b/elf/Makefile
@@ -397,7 +397,7 @@
 	  | $(AWK) '($$7 ~ /^UND(|EF)$$/ && $$1 != "0:" && $$4 != "REGISTER") { print; p=1 } END { exit p != 0 }'
 
 # interp.c exists just to get this string into the libraries.
-CFLAGS-interp.c = -D'RUNTIME_LINKER="$(slibdir)/$(rtld-installed-name)"' \
+CFLAGS-interp.c = -D'RUNTIME_LINKER="$(rtlddir)/$(rtld-installed-name)"' \
 		  -DNOT_IN_libc=1
 $(objpfx)interp.os: $(common-objpfx)config.make
 
@@ -440,7 +440,7 @@
 endif
 
 
-common-ldd-rewrite = -e 's%@RTLD@%$(slibdir)/$(rtld-installed-name)%g' \
+common-ldd-rewrite = -e 's%@RTLD@%$(rtlddir)/$(rtld-installed-name)%g' \
 		     -e 's%@VERSION@%$(version)%g' \
 		     -e 's%@PKGVERSION@%$(PKGVERSION)%g' \
 		     -e 's%@REPORT_BUGS_TO@%$(REPORT_BUGS_TO)%g'
